pkg/coscheduling/coscheduling.go

```diff
@@ -282,45 +235,15 @@ func (cs *Coscheduling) PreFilter(ctx context.Context, state *framework.CycleSta
 	if len(pgKey) == 0 {
 		return framework.NewStatus(framework.Success, "")
 	}
-	pgMinAvailable := pgInfo.minAvailable
 
-	waitingTime := time.Now().Sub(cs.creation)
-	if _, ok := cs.encounteredGroups.Load(pgInfo.name); !ok {
-		klog.V(9).Infof("INFO: encountered a new pod group, refreshing info")
-		cs.getNewWaitingGroups()
-	} else if waitingTime > time.Duration(PodScheduleTimeout)*time.Second {
-		klog.V(9).Infof("INFO: waiting groups timed out, refreshing info")
+	if mapSize(cs.approvedGroups) == 0 {
```

Comment:
A race condition here, no?

Reply:
I don't think so? will change this slightly, but mapSize just uses the range function in maps, which shouldn't lock anything beyond itself. Reassigning to a new map just helps clear some leftover metadata that could take a lot of space if the scheduler runs for a long time.